mkosi/__init__.py

```diff
@@ -1902,31 +1903,54 @@ def reenable_kernel_install(args: CommandLineArguments, root: str) -> None:
     make_executable(hook_path)
 
 
+def add_packages(
+    args: CommandLineArguments, packages: Set[str], *names: str, conditional: Optional[str] = None
+) -> None:
+    """Add packages in @names to @packages, if enabled by --base-packages.
+
+    If @conditional is specifed, rpm-specific syntax for boolean
+    dependencies will be used to include @names on if @conditional is.
+    """
+    assert args.base_packages is True or args.base_packages is False or args.base_packages == "conditional"
+
+    if args.base_packages or (conditional or args.base_packages is True):
+        for name in names:
+            packages.add(f"({name} if {conditional})" if conditional else name)
+
+
+def sort_packages(packages: Set[str]) -> List[str]:
+    """Sorts packages: normal first, paths second, conditional third"""
+
+    m = {"(": 2, "/": 1}
```

Comment:
I think this only works for absolute paths, could this be an issue?

Reply:
At least for virtual deps, the path must be absolute, and this was the case I was thinking about.

I guess you can also specify an rpm by path, to install a local rpm instead of downloading it. Then it would be either sorted last (if an absolute path is given), and somewhere in the middle alphabetically if the path starts with letters. I think it's OK to ignore this corner case. The sort would still be stable.